Add page on sharing before shipping (#12937)
expo/expo: docs/pages/guides/sharing-before-release.md

---

> title: Sharing before release

Reviewer: i think this title is a bit unclear

Author: Do you have a suggestion?  The original title was "Sharing your app before you ship it to the stores" but that's pretty long

https://linear.app/expo/issue/WR-61/sharing-your-app-before-you-ship-it-to-the-stores

Reviewer: i think it's unclear because:
- "before release" could be read as "before the first time you submit to the store" 
- "Sharing" is not a word i think people associate with distributing test builds, it is strongly associated with social media sharing

maybe: "Creating and distributing test builds" or something

---

import ImageSpotlight from '~/components/plugins/ImageSpotlight'

> Ultimately, you will want to publish your application to the App Store and Play Store to access the opportunities for wider distribution and to provide a familiar installation process for your users.  While you are developing though, you'll want to let other members of your team and others with a vested interet in the project see the work in progress to gather feedback and make sure you're building a quality product.

Reviewer 2: ```suggestion
Ultimately, you will want to publish your application to the App Store and Play Store to access the opportunities for wider distribution and to provide a familiar installation process for your users.  While you are developing though, you'll want to let other members of your team and others with a vested interest in the project see the work in progress to gather feedback and make sure you're building a quality product.
```

> Ultimately, you will want to publish your application to the App Store and Play Store to access the opportunities for wider distribution and to provide a familiar installation process for your users.  While you are developing though, you'll want to let other members of your team and others with a vested interet in the project see the work in progress to gather feedback and make sure you're building a quality product.

Reviewer: ```suggestion
The typical means of distribution for a mobile app is through the Apple App Store and/or Google Play Store, but this isn't the best way to share pre-production build. During the development process you'll want share your progress with both technical and non-technical stakeholders in the project to gather feedback and test it, and there are several approaches you can take for doing this with Expo projects.
```

## Expo Go

> If you are using the Expo Managed workflow, you can share your application with your team through the Expo Go app.

Reviewer: should talk about why you might do this vs another approach

1. Publish the latest version of your project to Expo's servers by running `expo publish`.
2. [Invite your teammate](https://expo.io/[account]/[project]/settings/members) to the Expo account that owns the project.
3. Have them [download the Expo Go app](https://expo.io/expo-go) onto their device.
4. They'll be able to open your application from the Profile tab of their Expo Go app.

> <ImageSpotlight alt="Terminal running expo init, with minimal (TypeScript) selected" src="/static/images/qr-code.gif" style={{height: '300px', width: 'auto'}} />

Reviewer 2: video is an excellent addition 👍 but (as I found out just today) we should make this an mp4 (I have this file, can send it to you), and then use the Video component so that it matches the other screen recordings (like [this one](https://docs.expo.io/versions/latest/sdk/imagepicker/))

> ## Internal Distribution

Reviewer: can't we just give a quick explanation of what this lets you do and link to internal distribution docs? feels repetitive.

Author: The internal distribution docs explain how to accomplish sharing APKs, ad hoc or enterprise builds through EAS.  This page is geared towards people who don't understand what their options are

> If you want to share outside of Expo Go, Android and iOS both offer ways to install a build of your application directly on devices.

Reviewer: should address why

> ### on Android

Reviewer 2: ```suggestion
### Android
```

> To share your application to Android devices, you must build an APK of your project rather than an AAB that is optimized for distribution through the Play Store. APKs can be downloaded (from a browser, email, or chat app) and installed on any Android device once the user accepts the security risk of installing an app that has not gone through Play Store review.

Reviewer 2: ```suggestion
To share your application to Android devices, you must build an APK of your project rather than an AAB, which is optimized for distribution through the Play Store. APKs can be downloaded (from a browser, email, or chat app) and installed on any Android device once the user accepts the security risk of installing an app that has not gone through Play Store review.
```

> To share your application to Android devices, you must build an APK of your project rather than an AAB that is optimized for distribution through the Play Store. APKs can be downloaded (from a browser, email, or chat app) and installed on any Android device once the user accepts the security risk of installing an app that has not gone through Play Store review.

Reviewer: ```suggestion
There are two types of Android binaries: APKs (Android application package files) and AABs (Android app bundles). APKs can be installed directly to an Android device over USB, by downloading the file over the web or through an email or chat app, once the user accepts the security warning for installing an app that has not gone through Play Store review. AABs must be distributed through the Play Store.
```

### Ad Hoc Distribution on iOS

Apple offers [Ad Hoc provisioning profiles](https://help.apple.com/xcode/mac/current/#/dev7ccaf4d3c) to distribute your app to test devices once they have been registered to your Apple Developer account.  This method requires a paid Apple Developer account and that account will only be able to use this method to distribute to at most 100 iPhones per year.

You will need to know the UDID (Unique Device Identifier) of each device that will install your application, which may be challenging if you are trying to share with someone who is not a developer.  Adding a new device will require a rebuild of your application. 

> Setting up Ad Hoc certificates correctly can be intimidating if you haven't done it before.  If you're using [EAS Build](build/internal-distribution.md), which is optimized for Expo and React Native projects, we'll handle the time-consuming parts of setting up Ad Hoc credentials for you.

Reviewer: ```suggestion
Setting up Ad Hoc certificates correctly can be intimidating if you haven't done it before.  If you're using [EAS Build](/build/internal-distribution.md), which is optimized for Expo and React Native projects, we'll handle the time-consuming parts of setting up Ad Hoc credentials for you.
```

> Setting up Ad Hoc certificates correctly can be intimidating if you haven't done it before.  If you're using [EAS Build](build/internal-distribution.md), which is optimized for Expo and React Native projects, we'll handle the time-consuming parts of setting up Ad Hoc credentials for you.

Reviewer: ```suggestion
Setting up Ad Hoc certificates correctly can be intimidating if you haven't done it before. If you're using [EAS Build](build/internal-distribution.md), which is optimized for Expo and React Native projects, we'll handle the time-consuming parts of setting up Ad Hoc credentials for you.
```

Reviewer: `%s/\.\s\s/. /g` could be a useful to run on docs to clear out the `.  ` typo 😅

Reviewer: oh i was seeing an outdated diff, nm

### Enterprise Distribution on iOS

> If your app is only intended to be used internally and will not be distributed through the App Store, you should use Enterprise distribution.  Unlike Ad Hoc Distribution, there is no limit to the number of devices that can install your application, and you will not need to manage the UDIDs of each device.  Enterprise Distribution requires membership in the more expensive [Apple Developer Enterprise Program](https://developer.apple.com/programs/enterprise/), which requires your organization to be a legal entity and go through Apple's verification process. 

Reviewer: ```suggestion
If your app is only intended to be used internally and will not be distributed through the App Store, you should use Enterprise distribution. Unlike Ad Hoc Distribution, there is no limit to the number of devices that can install your application, and you will not need to manage the UDIDs of each device.  Enterprise Distribution requires membership in the more expensive [Apple Developer Enterprise Program](https://developer.apple.com/programs/enterprise/), which requires your organization to be a legal entity and go through Apple's verification process. 
```

## TestFlight

> TestFlight is another option to distribute your application to iOS devices.  TestFlight also requires a paid Apple Developer account, but allows you to share your application with up to 10,000 users that can be invited with their email or a public link.  This method requires you to [upload your application](/submit/ios.md) to App Store Connect and wait for the automated review before you can share a build.  If you intend to ship new builds frequently, investing the time to set up Ad Hoc distribution will be worthwhile.

Reviewer: ```suggestion
TestFlight is another option to distribute your application to iOS devices. TestFlight also requires a paid Apple Developer account, but allows you to share your application with up to 10,000 users that can be invited with their email or a public link.  This method requires you to [upload your application](/submit/ios.md) to App Store Connect and wait for the automated review before you can share a build.  If you intend to ship new builds frequently, investing the time to set up Ad Hoc distribution will be worthwhile.
```

---
title: Sharing before release
---

import Video from '~/components/plugins/Video'

> The typical means of distribution for a mobile app is through the Apple App Store and/or Google Play Store, but this isn't the best way to share pre-production build. During the development process you'll want share your progress with both technical and non-technical stakeholders in the project to gather feedback and test it, and there are several approaches you can take for doing this.

Reviewer 2: ```suggestion
The typical means of distribution for a mobile app is through the Apple App Store and/or Google Play Store, but this isn't the best way to share pre-production builds. During the development process you'll want share your progress with both technical and non-technical stakeholders of the project to gather feedback and test it, and there are several approaches you can take for doing this.
```

### Android

To share your application to Android devices, you must build an APK (Android application package file) of your project. APKs can be installed directly to an Android device over USB, by downloading the file over the web or through an email or chat app, once the user accepts the security warning for installing an app that has not gone through Play Store review. AAB (Android app bundle) binaries of your application must be distributed through the Play Store.

### Ad Hoc Distribution on iOS

Apple offers [Ad Hoc provisioning profiles](https://help.apple.com/xcode/mac/current/#/dev7ccaf4d3c) to distribute your app to test devices once they have been registered to your Apple Developer account. This method requires a paid Apple Developer account and that account will only be able to use this method to distribute to at most 100 iPhones per year.

You will need to know the UDID (Unique Device Identifier) of each device that will install your application, which may be challenging if you are trying to share with someone who is not a developer. Adding a new device will require a rebuild of your application. 

> Setting up Ad Hoc certificates correctly can be intimidating if you haven't done it before. If you're using [EAS Build](/build/internal-distribution.md), which is optimized for Expo and React Native projects, we'll handle the time-consuming parts of setting up Ad Hoc credentials for you.

Reviewer: ```suggestion
Setting up Ad Hoc certificates correctly can be intimidating if you haven't done it before, and tedious even if you have. If you're using [EAS Build](/build/internal-distribution.md), which is optimized for Expo and React Native projects, we'll handle the time-consuming parts of setting up Ad Hoc credentials for you.
```

### Enterprise Distribution on iOS

If your app is only intended to be used internally and will not be distributed through the App Store, you should use Enterprise distribution. Unlike Ad Hoc Distribution, there is no limit to the number of devices that can install your application, and you will not need to manage the UDIDs of each device. Enterprise Distribution requires membership in the more expensive [Apple Developer Enterprise Program](https://developer.apple.com/programs/enterprise/), which requires your organization to be a legal entity and go through Apple's verification process. 

> 

Reviewer: ```suggestion
```
